x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1453,15 +1465,19 @@ void computeHashForApiKey(SecureString apiKey, ActionListener<char[]> listener)
 
     // Protected instance method so this can be mocked
     protected void verifyKeyAgainstHash(String apiKeyHash, ApiKeyCredentials credentials, ActionListener<Boolean> listener) {
-        threadPool.executor(SECURITY_CRYPTO_THREAD_POOL_NAME).execute(ActionRunnable.supply(listener, () -> {
-            Hasher hasher = Hasher.resolveFromHash(apiKeyHash.toCharArray());
-            final char[] apiKeyHashChars = apiKeyHash.toCharArray();
-            try {
-                return hasher.verify(credentials.getKey(), apiKeyHashChars);
-            } finally {
-                Arrays.fill(apiKeyHashChars, (char) 0);
-            }
-        }));
+        try {
+            threadPool.executor(SECURITY_CRYPTO_THREAD_POOL_NAME).execute(ActionRunnable.supply(listener, () -> {
+                Hasher hasher = Hasher.resolveFromHash(apiKeyHash.toCharArray());
+                final char[] apiKeyHashChars = apiKeyHash.toCharArray();
+                try {
+                    return hasher.verify(credentials.getKey(), apiKeyHashChars);
+                } finally {
+                    Arrays.fill(apiKeyHashChars, (char) 0);
+                }
+            }));
+        } catch (RejectedExecutionException e) {
+            listener.onFailure(e);
```

Comment:
Should we re-interpret this into something that'll come out as a 429? I'm not 100% sure which kind of exception that is off the top, I'll look into it.

Reply:
If you're executing an `AbstractRunnable` then it'll never throw like this, it'll call `AbstractRunnable#onRejection` (which ends up calling `listener#onFailure` by default, tho in this case maybe it'd be better to override that rather than to catch the rejection further up?)

Reply:
I don't quite understand, because we're submitting an `ActionRunnable`, which is a subclass of `AbstractRunnable`, but we still definitely hit this `catch`. So can you clarify what you mean by "it'll never throw like this"?

Reply:
Err, wait, I think I see. So maybe we just need the change in caching, and not this try/catch. Let me try that.

Reply:
Yep don't need this try/catch, good feedback, thanks very much for the review!